bson/bsonrw/extjson_reader.go

```diff
@@ -164,6 +164,19 @@ func (ejvr *extJSONValueReader) skipObject() {
 	depth := 1
 	for depth > 0 {
 		ejvr.p.advanceState()
+
+		// If object is empty, raise depth and continue. If there is a comma, there are
+		// remaining fields, emptyObject must be set back to false, and comma must be
+		// skipped with advanceState().
+		if ejvr.p.emptyObject {
```

Comment:
Optional: Add a description of the expected state of the parser/buffer to help people reading the code understand why we need different behavior for when `ejvr.p.emptyObject` is true.

E.g.
> When emptyObject is true, the parser has already read both the opening and closing brackets of an empty object ("{}"), so the next valid token will be part of the parent document, not part of the nested document.

Reply:
Good idea! Added a description of the state of the parser (the exact text you gave).